### ISSUE TITLE
interesting visualization effect/bug

### ISSUE THREAD

**Issue by RSly** [NVIDIA/DIGITS#1401]
Hi
I have defined a python layer to calculate a new loss, all works fine but I get these strange curves

it looks like there is a bug in the visualization? or maybe the loss is sometimes just not calculated?

I use exclude{stage: "deploy"  }  for these layers.

![image](https://cloud.githubusercontent.com/assets/3428292/22066524/9075171c-dd8d-11e6-9ad8-a9390377d1f7.png)

**Comment by RSly** [comment]
I figured out that those wholes are maybe caused by none values
is there a way to see such cases for better debugging?

**Comment by gheinrich** [comment]
Can you attached your `caffe_output.log`? I think I've seen similar things in cases where there was too much output coming out of Caffe.

**Comment by lukeyeager** [comment]
> I figured out that those wholes are maybe caused by none values

Yep, probably.

FWIW, I actually think that DIGITS and the graphing library is handling this pretty well!

**Comment by RSly** [comment]
@gheinrich good idea to check the caffe_output.log
Actually it seems that the since I am printing out some values from my python script, this simply makes the above graph not to be drawn currently!!

So I commented my "print loss" line in the python script and the graph is now whithout the wholes

Is it the behavior we want from DIGITS?
attached is the log file
[caffe_output (1).txt](https://github.com/NVIDIA/DIGITS/files/716515/caffe_output.1.txt)

**Comment by gheinrich** [comment]
Yes, the things that you print during training (either in your script or in Caffe) can confuse the log parser in DIGITS so use debug prints with care.

**Comment by RSly** [comment]
Thanks @gheinrich 
by the way, do you have any idea why my loss is increasing (the blue curve)?
also my training loss is much higher than my val loss (green curve)

I am trying to implement the dice loss with the backward included.
I couldn't find the backward implementation from your scripts

**Comment by gheinrich** [comment]
> do you have any idea why my loss is increasing (the blue curve)?
> also my training loss is much higher than my val loss (green curve)

It is difficult to say... it's always good to check your data.

> I am trying to implement the dice loss with the backward included.
> I couldn't find the backward implementation from your scripts

There is an implementation of the backward pass for Dice there:
https://github.com/faustomilletari/VNet/blob/master/pyLayer.py#L55-L66

**Comment by RSly** [comment]
Thanks, yes I am using a similar implementation.
it is confusing at this time :)
